p2p/peerserver/banpeerhandler.go

```diff
@@ -56,9 +61,15 @@ func (sp *serverPeer) addBanScore(persistent, transient uint32, reason string) {
 	if score > warnThreshold {
 		log.Warn(fmt.Sprintf("Misbehaving peer %s: %s -- ban score increased to %d",
 			sp, reason, score))
-		if score > sp.server.cfg.BanThreshold {
+		if score >= sp.server.cfg.BanThreshold {
 			log.Warn("Misbehaving peer -- banning and disconnecting", "peer", sp)
-			sp.server.BanPeer(sp)
+			dur := float64(transient) / float64(sp.server.cfg.BanThreshold)
+			dur *= float64(sp.server.cfg.BanDuration)
+			msg := BanPeerMsg{sp: sp, dur: time.Duration(dur)}
+			if msg.dur > sp.server.cfg.BanDuration {
```

Comment:
I think ban threshold/ban duration definitions should be refactored and stay together with ban score definitions, so that code reader can be clear to know the `defaultBanThreshold` is 100, and `warnThreshold` is 50, and the `SeriousScore` is 100, the `BanDuration` is 24h. It is more maintainable for future changes

Reply:
https://github.com/Qitmeer/qitmeer/pull/255/commits/b8470458f82c4b88dfa42aa2e84fa3b7026b8f20